pkg/session/session.go

```diff
@@ -3815,10 +3815,13 @@ func (s *session) PrepareTSFuture(ctx context.Context, future oracle.Future, sco
 		future = txnFailFuture{}
 	})
 
+	pipelined := s.sessionVars.BulkDMLEnabled && s.sessionVars.IsAutocommit() && !s.sessionVars.InTxn() &&
```

Comment:
Is this the only entry that auto-commit transaction would be activiated? Is the prepared protocol execution path also covered here?

Reply:
Added test for positive & negative cases.